### PR TITLE
Update to be simpler for re-inclusion into seabird

### PR DESCRIPTION
**Pull request by belak**
Quite a few changes. Note that when this gets re-included in seabird, we'll need to add a ping loop there if it doesn't already exist.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -9,257 +9,118 @@ import (
...
 	currentNick string
-	nick        string
-	user        string
-	name        string
-	password    string
-	lock        *sync.Mutex
 	conn        io.ReadWriteCloser
-	t           *tomb.Tomb
-	write       chan string
+	in          *bufio.Reader
```
Comment (jsvana) [inline client.go]: :D

File: client.go [hunk 2]
```diff
@@ -9,257 +9,118 @@ import (
...
 
-	c.connected = true
+	c.Writef("NICK %s", nick)
+	c.Writef("USER %s 0.0.0.0 0.0.0.0 :%s", user, name)
 
-	return c.start()
+	return c
+}
+
+func (c *Client) CurrentNick() string {
```
Comment (jsvana) [inline client.go]: Why not just expose currentNick?

**Comment by jsvana** [comment]
Looks great!

File: client.go (hunk 2 above)
Comment (belak) [inline client.go]: The idea is that we want it read only. But I suppose if it's documented it would be alright?

**Pull request merged**